autogpt_platform/backend/schema.prisma

```diff
@@ -277,3 +278,41 @@ model UserBlockCredit {
 
   @@id(name: "creditTransactionIdentifier", [transactionKey, userId])
 }
+
+enum APIKeyPermission {
+  EXECUTE_GRAPH // Can execute agent graphs
+  READ_GRAPH // Can get graph versions and details
+  EXECUTE_BLOCK // Can execute individual blocks
+  READ_BLOCK // Can get block information
+}
+
+model APIKey {
+  id          String             @id @default(uuid())
+  name        String
+  prefix      String             @unique // First 8 chars for identification
+  key         String             @unique // Hashed key
```

Comment:
If we're hashing the API key, do we need a column to store the salt? Or is it included in this column as a suffix?

Reply:
For API keys, typically don't need a salt because:

An API key is already:

Randomly generated
Unique
High entropy (using token_urlsafe)
Long enough to be secure

unlike passwords, where you would want a salt

Reply:
https://auth0.com/blog/adding-salt-to-hashing-a-better-way-to-store-passwords/

A salt has little benefit here imo as if the keys themselves are high entropy it's unlikely they are in hash and rainbow tables. If this is a concern, yes it would need a column but beware that you need to pick an algorithm that allows for len(salt) + len(key) to be within its range. [Bcrypt](https://security.stackexchange.com/questions/39849/does-bcrypt-have-a-maximum-password-length) for example has a max of 72 bytes (keep in mind 💩is 4 bytes when counting) so that will need checked too.